insrequester/requester.go

```diff
@@ -103,25 +104,19 @@ func (r *Request) sendRequest(httpMethod string, re RequestEntity) (*http.Respon
 		}
 
 		if res.StatusCode >= http.StatusInternalServerError {
-			outerErr = err
-			return inscodeerr.CodeErr{
-				Code:    res.StatusCode,
-				Message: fmt.Sprintf("status code: %v", res.StatusCode),
-			}
+			err := errors.New(fmt.Sprintf("status code: %v", res.StatusCode))
+			return err
 		}
 
-		if res.StatusCode < http.StatusOK || res.StatusCode >= http.StatusMultipleChoices {
-			outerErr = inscodeerr.CodeErr{
-				Code:    res.StatusCode,
-				Message: fmt.Sprintf("status code: %v", res.StatusCode),
-			}
-			return nil
-		}
```

Comment:
400'leri retryable ve circuit-breakable error olmaktan çıkarıyor muyuz, neden?

Reply:
4xx'leri neden hata olarak atmamaliyiz?

Reply:
bu app logici değil mi, ilk elden neden ekledik zaten

Reply:
4xx durumlarini retry etmeye gerek yok gibi. bunun gerektigi bi use case var mi ornek

Reply:
xml parser config'inden 
![image](https://github.com/useinsider/go-pkg/assets/3155556/0b896c8c-098e-4a19-b53f-a8705295a2c5)

Reply:
bu değişikliğe ihtiyaç nereden doğdu?

Reply:
runner icinde codeerr return edince parse edilemiyor error. hatayi disarda cozumleyemiyoruz. sonra 4xxlerin hata olmarak donmemesi gerektigini farkettim, disarda kontrol ediliyo bunlar

Reply:
xml parser a uygun bu, her yere uygun olmayabilir değil mi? Mümkün olduğunca general yazmaya çalışıyoruz kütüphaneleri

Reply:
evet mantıklı

Reply:
@sezaakgun senin dediğin gibi genel yazmak deyince, axios'un retryable codes config'ini örnek almak lazım
![image](https://github.com/useinsider/go-pkg/assets/3155556/b786884f-7f90-479f-a61a-c326419a831d)

Reply:
zaten 500'u de disarda handle edebiliyor isterse. response icinde bu 5xx bilgisi var

Reply:
bu sendRequest'ten dondugumuz err, requestten gelen 5xx 4xx ler olmamali. runner icind 5xx'leri err olarak return etmemizin sebebi retrier ve circuit breaker'in bunu yakalamasi

Reply:
bi axios'un status code'larini ekleyebilirim iceri

Reply:
ekledim

Reply:
okey o zaman öncesinde 5xx gelince response dönmüyoruk, artık outerErr tüm status code'lar için nil olduğundan tüm statuc code'larda res'e ulaşım var diye anlıyorum. o zaman müko.

diğer yandan, şu err'e verdiğimiz status code vb text'lere herhangi bir yerden ulaşılmıyor o zaman demek oluyor. burayı basitleştirsek mi? Dummy bir err return edelim. örn RetriableError diye bir const açalım.
```
err := errors.New(fmt.Sprintf("status code: %v", res.StatusCode))
```